src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/MetricsEmitter.java

```diff
@@ -847,4 +847,79 @@ public static void emitNodeMetrics(
           "Total time taken for writing {} metrics metricsdb: {}", tableName, mFinalT - mCurrT);
     }
   }
+
+
+  public static void emitShardStateMetric(
+          MetricsDB metricsDB, ShardStateMetricsSnapshot shardStateMetricsSnapshot) {
+    long mCurrT = System.currentTimeMillis();
+    Result<Record> shardStateMetrics = shardStateMetricsSnapshot.fetchAggregatedShardStateMetrics();
+    List<String> dims =
+            new ArrayList<String>() {
+              {
+                this.add(AllMetrics.ShardStateDimension.INDEX_NAME.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_ID.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_TYPE.toString());
+                this.add(AllMetrics.ShardStateDimension.NODE_NAME.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_STATE.toString());
+              }
+            };
+    metricsDB.createMetric(
+            new Metric<Double>(AllMetrics.ShardStateValue.SHARD_STATE.toString(), 0d),
+            dims);
+
+    BatchBindStep handle =
+            metricsDB.startBatchPut(
+                    new Metric<Double>(AllMetrics.ShardStateValue.SHARD_STATE.toString(), 0d),
+                    dims);
+
+    for (Record r : shardStateMetrics) {
+
+      Double sumShardState =
+              Double.parseDouble(
+                      r.get(
+                              DBUtils.getAggFieldName(
+                                      AllMetrics.ShardStateValue.SHARD_STATE.toString(),
+                                      MetricsDB.SUM))
+                              .toString());
+
+      Double avgShardState =
+              Double.parseDouble(
+                      r.get(
+                              DBUtils.getAggFieldName(
+                                      AllMetrics.ShardStateValue.SHARD_STATE.toString(),
+                                      MetricsDB.AVG))
+                              .toString());
+
+      Double minShardState =
+              Double.parseDouble(
+                      r.get(
+                              DBUtils.getAggFieldName(
+                                      AllMetrics.ShardStateValue.SHARD_STATE.toString(),
+                                      MetricsDB.MIN))
+                              .toString());
+
+      Double maxShardState =
+              Double.parseDouble(
+                      r.get(
+                              DBUtils.getAggFieldName(
+                                      AllMetrics.ShardStateValue.SHARD_STATE.toString(),
+                                      MetricsDB.MAX))
+                              .toString());
```

Comment:
Trying to understand why this is relevant or required. The shard states as I understand are enum/string values and we are trying to find an sum/avg/min/max over this string values ?